### ISSUE TITLE
Unable to produce pattern in lights (Valhelsia 1.16.5)

### ISSUE THREAD

**Issue by fightmesun** [pau101/Fairy-Lights#114]
when crafting multiple colours, the outcome will only take the first light. So if I make a pattern of red, black, purple; only a string of red lights will be crafted

**Comment by pau101** [comment]
Thanks for reporting!

Could you provide a screenshot?

**Comment by SpaceEngineerRaptor** [comment]
I've also experienced this issue, and I believe it to be related to mods such as FastWorkbench (I think that's its name) that change how crafting recipes are handled.
A short term fix is to take one of the vital ingredients (iron or string) out and put it back in, the recipe will temporarily become invalid when you do it and then when you put it back, it will re-look up the recipe and see the other ingredients, and update the output accordingly.

**Comment by copygirl** [comment]
https://github.com/Shadows-of-Fire/FastWorkbench/issues/72#issuecomment-809726113

> as per the fix for #58 [c2a1a64](https://github.com/Shadows-of-Fire/FastWorkbench/commit/c2a1a642f28618314fc18016d5a5690d8e5edfbd), FB now always pushes a stack update every time the input changes if either the recipe changes, or the recipe is dynamic. Given that this recipe does this, it sounds like it's not properly returning true to IRecipe#isDynamic?

It looks like, for example, the hanging lights recipe, creates a dynamic recipe with an output:

https://github.com/pau101/Fairy-Lights/blob/ab32fe7c52bda6e9894e7ac2ee1256b0e6fb2af3/src/main/java/me/paulf/fairylights/server/item/crafting/FLCraftingRecipes.java#L231

Which causes the `isDynamic` to return false:

https://github.com/pau101/Fairy-Lights/blob/ab32fe7c52bda6e9894e7ac2ee1256b0e6fb2af3/src/main/java/me/paulf/fairylights/util/crafting/GenericRecipe.java#L121-L123

**Comment by pau101** [comment]
Yes this correct, I have recipes which `isDynamic()` returns false, as they implement `getIngredients()` for an example recipe so the recipe book is capable of displaying them. It looks like FastWorkbench is constructing its own exceptions of the semantics of the method. Nothing I can do I my end.

**Comment by copygirl** [comment]
Issue has been fixed on FastWorkbench's side.
